app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -516,6 +438,7 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
                 saveAnswersForCurrentScreen(DO_NOT_EVALUATE_CONSTRAINTS);
                 break;
             case HIERARCHY_ACTIVITY:
+            case HIERARCHY_ACTIVITY_FIRST_START:
```

Comment:
Needed for refreshing the UI when you open the hierarchy view for a saved for and select a question to jump to.